lib/ddtrace/metrics.rb

```diff
@@ -18,8 +18,9 @@ def initialize(options = {})
     end
 
     def supported?
-      version = Gem.loaded_specs['dogstatsd-ruby'] \
-                  && Gem.loaded_specs['dogstatsd-ruby'].version
+      version = (Gem.loaded_specs['dogstatsd-ruby'] \
+                  && Gem.loaded_specs['dogstatsd-ruby'].version) \
+                  || (defined?(Datadog::Statsd::VERSION) && Gem::Version.new(Datadog::Statsd::VERSION))
```

Comment:
```suggestion
      version = defined?(Datadog::Statsd::VERSION) && Gem::Version.new(Datadog::Statsd::VERSION) ||
        Gem.loaded_specs['dogstatsd-ruby'] && Gem.loaded_specs['dogstatsd-ruby'].version
```

I don't think that the order we check these two possible conditions matters, but `defined?` is much faster than `Gem.loaded_specs`:

```ruby
require 'benchmark/ips'

Benchmark.ips do |x|
  x.config(time: 2, warmup: 0.1)

  x.report('defined?', 'defined?(Datadog::Statsd::VERSION)')
  x.report('loaded_specs', "Gem.loaded_specs['dogstatsd-ruby']")

  x.compare!
end

# Comparison:
#            defined?: 69599614.9 i/s
#        loaded_specs: 13815010.0 i/s - 5.04x  (± 0.00) slower
```

Reply:
makes sense, updated